cmd/gateway/gateway.go

```diff
@@ -0,0 +1,25 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gateway
+
+import (
+	// Import all gateways.
+	_ "github.com/minio/minio/cmd/gateway/azure"
+	_ "github.com/minio/minio/cmd/gateway/b2"
+	_ "github.com/minio/minio/cmd/gateway/gcs"
+	_ "github.com/minio/minio/cmd/gateway/s3"
```

Comment:
Why can't we import whichever needed than blank imports?

This go file should have common code/definitions across all gateways.

Reply:
Blank imports are to run the init() which would register each commands. So all future gateways get added in this fashion and become part of the build

Reply:
Let me know if you have a better suggestion.